### PR TITLE
[SPARK-10079][SPARKR] Make 'column' and 'col' functions be S4 functions.

### PR DESCRIPTION
**Pull request by sun-rui**
1.  Add a "col" function into DataFrame.
2.  Move the current "col" function in Column.R to functions.R, convert it to S4 function.
3.  Add a s4 "column" function in functions.R.
4.  Convert the "column" function in Column.R to S4 function. This is for private use.

### CODE REVIEW

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -946,15 +946,36 @@ setMethod("foreachPartition",
...
+#' @aliases col
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlContext <- sparkRSQL.init(sc)
+#' df <- createDataFrame(sqlContext, list(50, 60, 70), "age")
+#' col(df, "age")
+#'}
+setMethod("col",
```
Comment (shivaram) [inline R/pkg/R/DataFrame.R]: Do we need this method given that we have `df$colname` and `[[` as well ?

Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: df$colname and [[ actually are based on this function.

Comment (shivaram) [inline R/pkg/R/DataFrame.R]: Yeah but in terms of exporting this I'm not sure we need another method for the users ?

File: R/pkg/R/DataFrame.R [hunk 2]
```diff
@@ -946,15 +946,36 @@ setMethod("foreachPartition",
...
+#'
+#' Selects column based on the column name and return it as a Column.
+#' Note that the column name can also reference to a nested column like `a.b`.
+#'
+#' @param x A SparkSQL DataFrame
+#'
+#' @rdname select
+#' @name col
+#' @aliases col
+#' @export
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: Please add
`#' @family subsetting functions`

File: R/pkg/R/DataFrame.R (hunk 1 above)
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: `col` in R seems to be unrelated to this.
https://stat.ethz.ch/R-manual/R-devel/library/base/html/col.html
`Returns a matrix of integers indicating their column number in a matrix-like object, or a factor of column labels.`

Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: col is a API method of Scala DataFrame, my intention to add it in SparkR is for API parity. But if it conflicts with existing function in R base package, and we have alias functions ($ and [[]]), I agree that we can roll such change back.

**Comment by sun-rui** [comment]
@shivaram, any more comments? could we merge it now?

**Comment by shivaram** [comment]
LGTM. Merging this
